src/app/generations/[id]/page.tsx

```diff
@@ -21,6 +22,25 @@ export async function generateMetadata({
   };
 }
 
-export default function Page() {
+export default async function Page({
+  params,
+}: {
+  params: { id: string };
+}) {
+  try {
+    const image = await db
+      .selectFrom("generatedImages")
+      .select(["promptText"])
+      .where("id", "=", params.id)
+      .executeTakeFirst();
+
+    if (image?.promptText) {
+      const encoded = encodeURIComponent(image.promptText);
+      redirect(`/?prompt=${encoded}`);
+      return;
+    }
+  } catch (e) {
+    console.error("Error fetching prompt", e);
+  }
   redirect("/");
 }
```

Comment:
this should still be a client side redirect. we should fetch it from the backend on page load and then redirect